### ISSUE TITLE
Uncaught RangeError on loading a local file

### ISSUE THREAD

**Issue by alex40724** [steffest/BassoonTracker#63]
When dropping a local .mod file (previously exported by the tracker), I get a 

Uncaught RangeError: Offset is outside the bounds of the DataView
    at DataView.getUint32

error (Firefox and Chrome on MacOs 10.15).

<img width="1006" alt="Bildschirmfoto 2021-01-03 um 17 06 17" src="https://user-images.githubusercontent.com/1221841/103483348-f13c2680-4de6-11eb-84e2-2efa73231cb9.png">

Thanks for the great tool!

**Comment by alex40724** [comment]
[demo.zip](https://github.com/steffest/BassoonTracker/files/5762064/demo.zip)

This is the file. Other files seem to work. I also noticed that the "Browse" button in the local file dialog only appears in the "dev.html" version, not only in the regular "index.html" loading the minified js (current master branch, running on local apache).

**Comment by steffest** [comment]
Thanks for the ZIP.
Indeed, there are some invalid sample lengths in that file causing BassoonTracker to try read beyond the end of the file.
Both ProTracker and Fasttracker also don't load it.
At least BassoonTracker should display a message in such cases in stead of just failing silently.
I'll mark it as bug and fix it soon.
Thanks for the report!

**Comment by alex40724** [comment]
Thanks for the quick feedback.

Please note that the file has been exported by BassoonTracker itself. So trying to re-load the mod is the end of a longer process. If the invalid lengths are part of the sample files, the best time to show such a message would probably be when the sample is loaded the first time.

**Comment by steffest** [comment]
I fixed the underlying issue. 
When switching from XM to MOD, it still was possible to save more than 31 samples in a MOD file, which resulted in a corrupt file.
I added some validation and warning dialogs, also for samples that are too long for MOD format and would be truncated on save.

I case you still need it, this is your fixed demosong:
[demosong1_fixed.zip](https://github.com/steffest/BassoonTracker/files/6002648/demosong1_fixed.zip)
